File: packages/app/src/services/software/definitions/variants/javaSdk24.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'java-sdk-24';
	public name = 'Java SE Development Kit 24.x';
	public category = [SoftwareCategory.Development, SoftwareCategory.Runtime];
	public downloadName = 'jdk-24_windows-x64_bin.msi';
	public icon = 'java.png';
	public homepage = 'https://oracle.com/java/technologies/downloads';
	public parent = 'java-sdk';

	public async resolveDownloadUrl() {
		return ok('https://download.oracle.com/java/24/latest/jdk-24_windows-x64_bin.msi');
	}
}

File: src/lib/software/variants/java_sdk_24.py
from src.lib.software import BaseSoftware

class JavaSEDevelopmentKit24(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'java-sdk-24'
        self.name = 'Java SE Development Kit 24.x'
        self.download_name = 'jdk-24_windows-x64_bin.msi'
        self.icon = 'java.png'
        self.homepage = 'https://oracle.com/java/technologies/downloads'

    def resolve_download_url(self):
        self.url_resolved.emit('https://download.oracle.com/java/24/latest/jdk-24_windows-x64_bin.msi')
